src/deckgl-layers/layer-utils/cpu-aggregator.js

```diff
@@ -0,0 +1,606 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* eslint-disable guard-for-in */
+import {experimental, AGGREGATION_OPERATION} from '@deck.gl/aggregation-layers';
+import {aggregate} from 'utils/aggregate-utils';
+import {AGGREGATION_TYPES, SCALE_FUNC} from 'constants/default-settings';
+
+const {BinSorter} = experimental;
+
+export const DECK_AGGREGATION_MAP = {
+  [AGGREGATION_OPERATION.SUM]: AGGREGATION_TYPES.sum,
+  [AGGREGATION_OPERATION.MEAN]: AGGREGATION_TYPES.average,
+  [AGGREGATION_OPERATION.MIN]: AGGREGATION_TYPES.minimum,
+  [AGGREGATION_OPERATION.MAX]: AGGREGATION_TYPES.maximum
+};
+
+export function getValueFunc(aggregation, accessor) {
+  const op =
+    AGGREGATION_OPERATION[aggregation.toUpperCase()] ||
```

Comment:
should we have a default value here?
would it be better to throw an exception or something when a value is not part of our operations?

Reply:
👍